tools/ob_admin/backup_tool/ob_admin_backup_validation_task.cpp

```diff
@@ -0,0 +1,4804 @@
+    ret = OB_ERR_UNEXPECTED;
+    STORAGE_LOG(WARN, "user tablet id size not equal major tablet map size", K(ret),
+                K(backup_set_attr->user_tablet_id_.count()),
+                K(backup_set_attr->major_tablet_map_.size()));
+  } else {
+    // cross check tablet, found any may be missing
+    for (int64_t i = 0; OB_SUCC(ret) && i < backup_set_attr->user_tablet_id_.count(); i++) {
+      ObAdminBackupTabletValidationAttr *minor_tablet_attr = nullptr;
+      ObAdminBackupTabletValidationAttr *major_tablet_attr = nullptr;
+      if (OB_FAIL(backup_set_attr->minor_tablet_map_.get_refactored(
+              backup_set_attr->user_tablet_id_.at(i), minor_tablet_attr))) {
+        STORAGE_LOG(WARN, "failed to get minor tablet store", K(ret),
+                    K(backup_set_attr->user_tablet_id_.at(i)));
+      } else if (OB_FAIL(backup_set_attr->major_tablet_map_.get_refactored(
+                     backup_set_attr->user_tablet_id_.at(i), major_tablet_attr))) {
+        STORAGE_LOG(WARN, "failed to get major tablet store", K(ret),
+                    K(backup_set_attr->user_tablet_id_.at(i)));
+      } else if ((OB_ISNULL(minor_tablet_attr->tablet_meta_index_)
+                  && OB_ISNULL(major_tablet_attr->tablet_meta_index_))
+                 || (OB_NOT_NULL(minor_tablet_attr->tablet_meta_index_)
+                     && OB_ISNULL(major_tablet_attr->tablet_meta_index_))) {
+        // maybe deleted or lost
+        if (OB_FAIL(missing_tablet.push_back(
+                {minor_tablet_attr->ls_id_, backup_set_attr->user_tablet_id_.at(i)}))) {
+          STORAGE_LOG(WARN, "failed to push back tablet id", K(ret));
+        }
+      } else if (OB_ISNULL(minor_tablet_attr->tablet_meta_index_)
+                 && OB_NOT_NULL(major_tablet_attr->tablet_meta_index_)) {
+        // bug
+        ret = OB_ERR_UNEXPECTED;
+        STORAGE_LOG(WARN, "minor tablet meta should exist but got lost", K(ret));
+      }
+    }
+  }
+  if (OB_SUCC(ret) && missing_tablet.count() > 0) {
+    // query __all_backup_skipped_tablet_history
+    STORAGE_LOG(WARN, "missing tablet", K(ret), K(missing_tablet));
+    common::ObSqlString sql;
+    ObMySQLProxy::MySQLResult res;
+    sqlclient::ObMySQLResult *result = nullptr;
+    if (OB_ISNULL(ctx_->sql_proxy_)) {
+      // ret = OB_ERR_UNEXPECTED;
+      STORAGE_LOG(WARN, "sql proxy is nullptr, skip query check", K(ret));
+      if (OB_FAIL(ctx_->global_stat_.add_scheduled_tablet_count_(-missing_tablet.count()
+                                                                 * 2 /*major+minor*/))) {
+        STORAGE_LOG(WARN, "failed to add scheduled tablet count", K(ret));
+      }
+    } else {
+      FOREACH_X(missing_tablet_iter, missing_tablet, OB_SUCC(ret))
+      {
+        sql.reset();
+        res.reset();
+        result = nullptr;
+        if (OB_FAIL(sql.assign_fmt(
+                "SELECT * FROM %s WHERE tenant_id = %ld AND backup_set_id = %ld AND ls_id = %ld "
```

Comment:
not exec sql in ob_admin

Reply:
This is a TODO, should persist skipped tablet and no more rely on sys table